install_octopus.sh

```diff
@@ -1,66 +1,100 @@
 
-# exit on error and print each command
-set -xe
+
+
+# Function to display script usage
+usage() {
+  echo "Usage: $0 [--version <version_number>] [--download_dir <download_location>] [--install_dir <install_prefix>]"
+  echo "Options:"
+  echo "  --version <version_number>      Specify the version number of Octopus (e.g., 13.0, develop)"
+  echo "  --download_dir <download_location>   Specify the download location for Octopus source"
+  echo "  --install_dir <install_prefix>   Specify the install prefix for Octopus (default: /usr/local)"
+  echo "  -h, --help                      Display this help message"
+  exit 1
+}
+
+# Parse command-line arguments
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    --version)
+      version="$2"
+      shift
+      shift
+      ;;
+    --download_dir)
+      location="$2"
+      shift
+      shift
+      ;;
+    --install_dir)
+      prefix="$2"
+      shift
+      shift
+      ;;
+    -h|--help)
+      usage
+      ;;
+    *)
+      echo "Invalid option: $1"
+      usage
+      ;;
+  esac
+done
 
 # Check if the version number and location is provided
-if [ -z "$1" ]
-  then
-    echo "No version number provided"
-    exit 1
-else
-  version=$1
+if [ -z "$version" ]; then
+  echo "No version number provided"
+  usage
 fi
-if [ -z "$2" ]
-  then
-    echo "No download location provided"
-    exit 1
-else
-    location=$2
+
+if [ -z "$location" ]; then
+  echo "No download location provided"
+  usage
 fi
-if [ -z "$3" ]
-  then
-    echo "No install prefix provided using default location"
-    prefix=/usr/local
-else
-    prefix=$3
+
+if [ -z "$prefix" ]; then
+  echo "No install prefix provided using default location"
+  prefix="/usr/local"
 fi
 
+## MAIN ##
+# exit on error and print each command
+set -xe
 # make the location if it does not exist
-if [ ! -d $location ]; then
-  mkdir -p $location
-fi
-cd $location
+mkdir -p "$location"
+
+cd "$location"
 
 # if develop is provided, clone the main branch
 
 if [ $version == "develop" ]; then
   git clone https://gitlab.com/octopus-code/octopus.git .
 else
-    # download the tar file
-    wget https://octopus-code.org/download/${version}/octopus-${version}.tar.gz
-    tar -xvf octopus-${version}.tar.gz
-    mv octopus-$version/* .
-    rm -rf octopus-$version
-    # rm octopus-$version.tar.gz
+  # download the tar file
+  wget https://octopus-code.org/download/${version}/octopus-${version}.tar.gz
+  tar -xvf octopus-${version}.tar.gz
+  mv octopus-$version/* .
+  rm -rf octopus-$version
+  # rm octopus-$version.tar.gz
 fi
 
 date=$(date)
 
 # Record the version number and date
 if [ $version == "develop" ]; then
-    # Record which version we are using
-    git show > octopus-source-version
-    echo "octopus-source-clone-date: $date " >> octopus-source-version
+  # Record which version we are using
+  git log -1 --pretty=format:'%H %D' > octopus-source-version
```

Comment:
Un related to the main PR but is a small change so I added it here.
Record less information than before, only show current commit and branch name

Reply:
Okay.